examples/goldilocks_models/find_boundary.cc

```diff
@@ -0,0 +1,772 @@
+
+  cout << "Trail name: " << FLAGS_program_name << endl;
+  cout << "Git commit hash: " << endl;
+  std::system("git rev-parse HEAD");
+  cout << "Result of \"git diff-index HEAD\":" << endl;
+  std::system("git diff-index HEAD");
+  cout << endl;
+
+  const string dir = "../dairlib_data/goldilocks_models/find_boundary/robot_" +
+      to_string(FLAGS_robot_option) + "/";
+  if (!CreateFolderIfNotExist(dir)) return 0;
+
+  /*
+   * cout basic program information
+   */
+  cout << "\nBasic information:\n";
+  cout << FLAGS_program_name << endl;
+  int robot_option = FLAGS_robot_option;
+  int rom_option = FLAGS_rom_option;
+  cout << "robot_option: " << robot_option << endl;
+  cout << "rom_option: " << rom_option << endl;
+
+  /*
+   * initialize task space
+   */
+  cout << "\nInitialize task space:\n";
+  Task task;//Traj Opt tasks
+  SearchSetting search_setting;//Settings of searching the task space
+  if(robot_option==0)
+  {
+    task = Task({"stride length", "ground incline",
+                 "velocity"});
+    search_setting = SearchSetting(3,{"stride length", "ground incline",
+                                      "velocity"},{0.25,0,0.4},
+                                          {0.01,0.01,0.02});
+  }
+  else if(robot_option==1){
+    task = Task({"stride length", "ground incline",
+                 "velocity", "turning rate"});
+    search_setting = SearchSetting(4,{"stride length", "ground incline",
+                                      "velocity","turning rate"},
+                                   {0.3,0,0.5,0},{0.01,0.01,0.02,0.01});
+  }
+  //cout initial point information
+  int dim = 0;
+  for(dim = 0;dim<search_setting.task_dim();dim++)
+  {
+    cout<<"initial "<<search_setting.names()[dim]<<
+    ": "+to_string(search_setting.task_0()[dim])<<endl;
+  }
+
+  /*
+   * Iteration setting
+   */
+  cout << "\nIteration setting:\n";
+  cout<<"get nominal cost: "<<FLAGS_is_get_nominal<<endl;
+  cout<<"use optimized model: "<<FLAGS_use_optimized_model<<endl;
+  cout<<"optimized model index: "<<FLAGS_theta_index<<endl;
+  int max_iter = FLAGS_max_outer_iter;
+  //TODO:decide the threshold under different situation
+  double cost_threshold = 30;
+  if(robot_option==0)
+  {
+    if(FLAGS_is_get_nominal){
+      cost_threshold = 35;
+    }
+    else{
+      cost_threshold = 30;
+    }
+  }
+  else if(robot_option==1){
+    if(FLAGS_is_get_nominal){
+      cost_threshold = 35;
+    }
+    else{
+      cost_threshold = 30;
+    }
+  }
+  cout<<"cost_threshold "<<cost_threshold<<endl;
+
+  // create components for each dimension used to decide the direction
+  // considering we can only visualize 2D landscape, we only search within
+  // 2D space and fix other dimension
+  if(robot_option==0){
+    search_setting.SetExtendComponents("stride length",5,{0,-1,0.5,0.5,1});
+    search_setting.SetExtendComponents("ground incline",5,{0,-1,0.5,0.5,1});
+    search_setting.SetExtendComponents("velocity",1,{0});
```

Comment:
It doesn't look like we need the setter function `SetExtendComponents`. We can create "elements_" and pass it into the constructor at https://github.com/DAIRLab/dairlib/pull/171/files#diff-25ea7bf0927e455b7265fbe2b0032473R620 ?